### PR TITLE
Automate docs deployment

### PR DESCRIPTION
**Pull request by XhmikosR**
@mdo this is untested and we need you to add a `PERSONAL_TOKEN` in the repo secrets. It will run only when a tag is created, which I assume it's what we want?

Alternatively I could probably combine the deploy action with the test one but this might be simpler for now.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        run: npm ci
+
+      - name: Build the icons
+        run: npm run icons
+
+      - name: Build the docs
+        run: npm run docs-build
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v2
```
Comment (mdo) [inline .github/workflows/deploy.yml]: This has a v3 version—should we be using that?

Comment (XhmikosR) [reply .github/workflows/deploy.yml]: When I made the PR there wasn't a v3 :P I will have a look at the changes and adapt the PR.

Comment (XhmikosR) [reply .github/workflows/deploy.yml]: Actually, it might have been a typo after all 🤦‍♂

**Comment by XhmikosR** [comment]
Alright, I updated this, but we need to test it. Let me know how do you want to do it.

**Comment by mdo** [comment]
> EDIT: if we use a Personal Access Token, we can really limit who can push directly to the gh-pages branch. Which is a good thing AFAICT. But if you could ask internally, it would help us decide :)

I've added one to this repo, so we can try that if you'd like. My best bet is that we try this in another private repo maybe with a simplified build system. Thoughts?

**Comment by XhmikosR** [comment]
We could try it just after we make the alpha2 release and before we manually update the docs.

**Pull request merged**